### PR TITLE
Gamestate Inspector: replace scroll_label with scroll_text

### LINKED ISSUE #8032: Text in the gamestate inspector should be selectable and copyable
### Describe the desired feature

When I type `:inspect` to open the inspection console, it's often because I'm collaboratively debugging something with someone online, who would like to know a particular element of my gamestate. In such a case, it would be a lot easier to be able to simply copy and paste the text out from the inspector, instead of my current method where I have to take screenshots of it. I thought to file this during the most recent look I've been taking at cooljeanius/The_Earths_Gut#27 btw.

### PR DESCRIPTION
**Pull request by babaissarkar**
Resolves #8032.
Uses `scroll_text` instead of `scroll_label` so that text can be copied.
(Should not be squashed.)

_Known issue:_
Could face slight slowdown for large texts, but it won't crash while selecting text or lag during loading.

### CODE REVIEW

**Comment by ProditorMagnus** [comment]
Inspector has copy button, so if it change makes it twice as slow or twice as small, I would consider relying on button.

**Comment by cooljeanius** [comment]
I'm impressed that it's so simple!

**Review by Vultraz — APPROVED** [review]
If it works, no objections.

**Comment by babaissarkar** [comment]
ATM trying to debug what's causing the slowdown for large texts, so would like to merge after that. Alternatively, if that is acceptable, I can merge this and investigate the slowdown in another PR.

**Comment by Vultraz** [comment]
Handle slowdown separately. There’s a bunch of low-hanging fruit in the canvas parser, especially using `split_view` instead of `split`.

Changing the toggles should be fine, but I’d like to see what they look like.

**Comment by babaissarkar** [comment]
> Changing the toggles should be fine, but I’d like to see what they look like.

icon-collapse icon-expand

Screenshot from 2024-08-12 12-34-51

The icon actually will need more thought, since at this point they look too bright and kinda cartoony, so probably should leave this for now.

**Comment by babaissarkar** [comment]
With https://github.com/wesnoth/wesnoth/pull/9193/commits/3d7a820d79f9208b7199018ff0f51174e1b19e1c, the slowdown should be improved. Loading is almost instant, but it slows down when you try to select text in very large texts, especially towards the end of the text.

File: src/gui/widgets/multiline_text.hpp [hunk 1]
```diff
@@ -208,14 +209,19 @@ class multiline_text : public text_box_base
 	/** Line number of text */
 	unsigned line_num_;
 
+	bool init_cache_;
```
Comment (soliton-) [inline src/gui/widgets/multiline_text.hpp:212]: This is unused.

Comment (cooljeanius) [reply src/gui/widgets/multiline_text.hpp:212]: ok so I'm kind of curious, why does only the macOS Release build catch this?

Comment (babaissarkar) [reply src/gui/widgets/multiline_text.hpp]: fixed.

File: src/font/text.hpp [hunk 2]
```diff
@@ -235,6 +235,25 @@ class pango_text
...
+	 * Get a specific line from the pango layout
+	 *
+	 * @param index    the line number of the line to retrieve
+	 *
+	 * @returns        the PangoLayoutLine* corresponding to line number index
+	 */
+	PangoLayoutLine* get_line(int index);
+
+	/**
+	 * Given a byte index, fine out at which line the corresponding character
```
Comment (soliton-) [inline src/font/text.hpp:248]:
```suggestion
	 * Given a byte index, find out at which line the corresponding character
```

**Comment by CelticMinstrel** [comment]
> Another question: I am interested in changing the tree view's nodes to [+]/[-] images to indicate folding/unfolding. Will that be a good idea or I should leave them alone?

I'd prefer to use 🞂 for folded and 🞃 for unfolded, personally. Regardless, it's a good idea to add indicators of the state.

**Comment by babaissarkar** [comment]
> I'd prefer to use 🞂 for folded and 🞃 for unfolded, personally. Regardless, it's a good idea to add indicators of the state.

I can just adjust treeview to use the default definition

**Comment by Vultraz** [comment]
Just a doxygen error left.

**Comment by babaissarkar** [comment]
> Just a doxygen error left.

I'll rebase and merge it once the CI checks succeed.

File: src/font/text.hpp (hunk 2 above)
Comment (babaissarkar) [reply src/font/text.hpp]: fixed.

**Comment by Vultraz** [comment]
LGTM 🎉

**Pull request merged**